### PR TITLE
Blockchain package and base traits

### PR DESCRIPTION
**Pull request by Christewart**
The pull request implements the blockchain, blockchain serialization, and merkle tree calculation.

### CODE REVIEW

**Comment by Christewart** [comment]
ACK

File: src/main/scala/org/bitcoins/core/serializers/RawBitcoinSerializer.scala [hunk 1]
```diff
@@ -11,31 +11,27 @@ trait RawBitcoinSerializer[T] extends RawBitcoinSerializerHelper {
...
    * Reads a hexadecimal value and transforms it into the native
    * scala type T
- *
    * @param hex
    * @return
    */
   def read(hex : String) : T = read(BitcoinSUtil.decodeHex(hex))
 
   /**
    * Reads in bytes and transforms it into the approriate scala type T
```
Comment (TomMcCabe) [inline src/main/scala/org/bitcoins/core/serializers/RawBitcoinSerializer.scala]: Typo, again line 27

File: src/main/scala/org/bitcoins/core/protocol/blockchain/ChainParams.scala [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+  override def genesisBlock = createGenesisBlock(1296688602, 414098458, 0x1d00ffff, 1, Bitcoins(50))
+
+  override def requireStandardTransaction = true
+
+  /**
+    * Base58 prefixes for various keys/hashes on the network
+    * See: https://en.bitcoin.it/wiki/List_of_address_prefixes
+    * @return
+    */
+  override def base58Prefixes : Map[Base58Type,Seq[Byte]] = Map(
```
Comment (TomMcCabe) [inline src/main/scala/org/bitcoins/core/protocol/blockchain/ChainParams.scala:149]: Nitpicky: should the "Map" called in the base58Prefixes functions for mainNet/testNet look similar? I.e. for mainNet, it's on next line, but for testNet it's on same line.

**Comment by TomMcCabe** [comment]
ACK.

**Pull request merged**